src/MagnumPlugins/DrFlacAudioImporter/DrFlacImporter.cpp

```diff
@@ -44,42 +84,107 @@ auto DrFlacImporter::doFeatures() const -> Features { return Feature::OpenData;
 bool DrFlacImporter::doIsOpened() const { return _data; }
 
 void DrFlacImporter::doOpenData(Containers::ArrayView<const char> data) {
-    drflac* handle = drflac_open_memory(data.data(), data.size());
+
+    std::unique_ptr<drflac, DrFlacDeleter> handle(drflac_open_memory(data.data(), data.size()));
     if(!handle) {
         Error() << "Audio::DrFlacImporter::openData(): failed to open and decode FLAC data";
         return;
     }
 
     uint64_t samples = handle->totalSampleCount;
-    uint32_t frequency = handle->sampleRate;
     uint8_t numChannels = handle->channels;
     uint8_t bitsPerSample = handle->bitsPerSample;
 
-    _frequency = frequency;
-
-    if(numChannels == 1 && bitsPerSample == 8)
-        _format = Buffer::Format::Mono8;
-    else if(numChannels == 1 && bitsPerSample == 16)
-        _format = Buffer::Format::Mono16;
-    else if(numChannels == 2 && bitsPerSample == 8)
-        _format = Buffer::Format::Stereo8;
-    else if(numChannels == 2 && bitsPerSample == 16)
-        _format = Buffer::Format::Stereo16;
-    else {
+    // FLAC supports any bitspersample from 4 to 64, but DrFlac always gives us 32-bit samples
+    // So we normalize bit amounts to multiples of 8, rounding up
```

Comment:
@mosra I'll do this, but I would consider this a bad coding style choice: /* */ comments can't nest, which makes it very difficult to comment out large blocks of code while testing. I've encountered this a few times so far.